onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.h

```diff
@@ -12,39 +12,51 @@ class Graph;
 class Node;
 
 namespace QDQ {
+
+// Struct to represent a DQ->Op->Q node group
+struct NodeGroup {
+  std::vector<NodeIndex> dq_nodes;
+  std::vector<NodeIndex> q_nodes;
+  NodeIndex target_node;
+};
+
 // Base QDQ checker. Finds and provides the DQ and Q nodes to the operator specific checkers, as the QDQ optimizations
 // always involve those nodes.
 class BaseSelector : public NodeSelector {
  public:
-  bool Select(Graph& graph, const Node& node, std::unique_ptr<NodesToOptimize>& selection) const override;
+  bool Select(const GraphViewer& graph_viewer, const Node& node,
+              std::unique_ptr<NodesToOptimizeIndices>& selection) const override;
+
```

Comment:
Do we need to have two different Select methods here, or could we simply add a helper to convert from NodeGroup to NodesToOptimizeIndices as that seems like a general purpose thing to do?